docs/Developer Reference/selecting-data-feeds.md

```diff
@@ -9,32 +9,113 @@ excerpt: "Learn how to assess data feeds that you use in your smart contracts."
+
+Custom feeds have the following categories and compositions:
+
+- **On-chain single source feeds:** These feeds are similar to Off-chain Single Source Feeds. However, in this case the data source resides on-chain.
+- **On-chain Proof of Reserve Feeds:** Chainlink Proof of Reserve uses the largest decentralized collection of security-reviewed and Sybil-resistant node operators in the industry to acquire and verify reserve data, reserves in this use case reside on-chain.
+- **Technical Feeds:** Feeds within this category measure a particular technical metric from a specified blockchain. For example, Fast Gas or Block Difficulty.
+- **Total Value Locked Feeds:** These feeds measure the total value locked in a particular protocol.
+- **Custom Index Feeds:** An index calculates a function of the values for multiple underlying assets. The function is specific to that index and is typically calculated by our node operators following an agreed formula.
+
+If you plan on using one of these feeds and would like to get a more detailed understanding, please contact the Chainlink Labs team.
+
+### 🔴 Specialized Feeds
+
+These are highly complex and bespoke feeds that inherently contain more significant risk factors associated with them. Typical users of these feeds are our large institutional users with deep expertise in the market space they operate in.
+
+While these are monitored and well supported, they might not meet the same levels of resiliency as the above categories. We strongly advise you to speak with the team to understand their use case, properties, and associated risks.
+
+**Examples of Specialized feeds:**
+
+- **Off-chain Single Source Feeds:** Some data providers use a single data source, which might be necessary if only one source exists off-chain for a specific type of data.
+- **Off-chain Proof of Reserve Feeds:** Chainlink Proof of Reserve uses the largest decentralized collection of security-reviewed and Sybil-resistant node operators in the industry to acquire and verify reserve data, reserves in this use case reside on-chain.
+- **LP Token Feeds:** These feeds use a decentralized feed for the underlying asset as well as some custom calculations to value the LP tokens.
+- **Wrapped Calculated Feeds:** These feeds are typically pegged 1:1 to the underlying token or asset. Under normal market conditions, these feeds track its underlying value accurately. However, the price is a derivative formed from a calculated method and might not always track value precisely.
+
+> 📘 Note:
+>
+> Users should evaluate data providers to make sure they provide high-quality data that your smart contracts can rely on. Any error or omission in the provider's data might negatively impact your application and its users.
+
```

Comment:
```suggestion
```suggestion
At Chainlink Labs we plan for all eventualities, however small they might be. To help our users prepare for such events, we recommend adding additional tooling for custom or specialized feeds to protect their application/protocol. This tooling is put in place to mitigate against some of the following risks: extreme market events, possible malicious activity occurring on a 3rd party venues/contracts, potential delays, or outages.
Below are some of the examples of tooling some of our users have put in place:
- Circuit Breakers - In the case of an extreme price event, the contract would pause operations for a limited period of time.
- Contract update delays - contracts would not update until the protocol had received a recent fresh input from the data feed.
- Manual Kill Switch - Example: If a vulnerability/bug was discovered in one of the upstream contracts, The user would manually be able to cease operation and temporarily sever the connection to the Price feed.
- Additionally, monitoring - some users create their own monitoring alerts based on deviations in the price feeds they’re using.

```
```